rapids_build_backend/impls.py

```diff
@@ -17,6 +17,12 @@
 from .config import Config
 
 
+def _remove_rapidsai_from_config(config_settings):
```

Comment:
Could you put some of the context from the PR description into a docstring here? So that when we come across it again in the future, we'll remember why it exists.

Reply:
And while you're adding this and have all the context for the inputs and outputs, could you add type hints to help `mypy`?

```suggestion
def _remove_rapidsai_from_config(
    config_settings: typing.Union[dict[str, typing.Any], None]
) -> typing.Union[dict[str, typing.Any], None]:
```

Reply:
I've added the docstring and type annotations.